src/librustdoc/html/static/js/search.js

```diff
@@ -345,6 +345,7 @@ window.initSearch = function(rawSearchIndex) {
         // Check for type name and type generics (if any).
         function checkType(obj, val, literalSearch) {
             var lev_distance = MAX_LEV_DISTANCE + 1;
+            var tmp_lev = MAX_LEV_DISTANCE + 1;
```

Comment:
Why this `tmp_lev` change? Is it really necessary for the generics fix? IIUC, the `[NAME]` additions below are necessary, but I don't see why the `tmp_lev` addition is.

Reply:
True, I'll revert this one.

Reply:
Ah no, nevermind! So originally it was declared just below ([here](https://github.com/rust-lang/rust/pull/88268/files#diff-766c50b21af39c8c0abd3fdba5f3c49a73c38b7b66ded4c7939e83b7244dfd3dL385)), but since it's now being used [here](https://github.com/rust-lang/rust/pull/88268/files#diff-766c50b21af39c8c0abd3fdba5f3c49a73c38b7b66ded4c7939e83b7244dfd3dR413) as well, I needed to have the declaration at the top of the function.

Reply:
I don't really understand why the other changes were made. Like with https://github.com/rust-lang/rust/pull/88268#discussion_r711662034, can you please have someone who's more knowledgeable about rustdoc's JS review this part?

Reply:
I'll ask @Nemo157.